src/rulesRunner.js

```diff
@@ -6,42 +6,69 @@ import flatten from "flat";
 const { utils } = require("@rjsf/core");
 const { deepEquals } = utils;
 
-function doRunRules(
+function hasDiff(oldData, newData) {
```

Comment:
This is equivalent to !isEqual - I know we did some performance comparisons and I think switching deepequal out for this didn't improve anything - should we just use the deepequal util instead?

Reply:
We could do that